src/AVSGateway.test.js

```diff
@@ -0,0 +1,253 @@
+  );
+
+  let nullMessage;
+  await expect(unitUnderTest.sendTextMessageEvent(nullMessage)).rejects.toEqual(
+    expect.any(IllegalArgumentError)
+  );
+
+  await expect(unitUnderTest.sendTextMessageEvent(undefined)).rejects.toEqual(
+    expect.any(IllegalArgumentError)
+  );
+});
+
+test("that sending a testRequest throws if the accessToken empty or undefined.", async () => {
+  expect.assertions(3);
+
+  const userRequestToAlexa = "user request to Alexa";
+  const accessToken = "";
+
+  await expect(
+    unitUnderTest.sendTextMessageEvent(userRequestToAlexa, accessToken)
+  ).rejects.toEqual(expect.any(IllegalArgumentError));
+
+  let nullAccessToken;
+  await expect(
+    unitUnderTest.sendTextMessageEvent(userRequestToAlexa, nullAccessToken)
+  ).rejects.toEqual(expect.any(IllegalArgumentError));
+
+  await expect(
+    unitUnderTest.sendTextMessageEvent(userRequestToAlexa, undefined)
+  ).rejects.toEqual(expect.any(IllegalArgumentError));
+});
+
+it("handles gracefully if AVS returns INVALID_REQUEST_EXCEPTION error code", () => {
+  testErrorResponseFromAVSHandling("INVALID_REQUEST_EXCEPTION", 400);
+});
+
+it("handles gracefully if AVS returns UNAUTHORIZED_REQUEST_EXCEPTION error code", () => {
+  testErrorResponseFromAVSHandling("UNAUTHORIZED_REQUEST_EXCEPTION", 403);
+});
+
+it("handles gracefully if AVS returns UNSUPPORTED_MEDIA_TYPE error code", () => {
+  testErrorResponseFromAVSHandling("UNSUPPORTED_MEDIA_TYPE", 415);
+});
+
+it("handles gracefully if AVS returns THROTTLING_EXCEPTION error code", () => {
+  testErrorResponseFromAVSHandling("THROTTLING_EXCEPTION", 429);
+});
+
+it("handles gracefully if AVS returns INTERNAL_SERVICE_EXCEPTION error code", () => {
+  testErrorResponseFromAVSHandling("INTERNAL_SERVICE_EXCEPTION", 500);
+});
+
+it("handles gracefully if AVS returns N/A error code", () => {
+  testErrorResponseFromAVSHandling("N/A", 503);
+});
+
+it("handles gracefully if AVS returns an unexpected error code", async () => {
+  const userRequestToAlexa = "user request to Alexa";
+  const access_token = "a mock access_token";
+  const errorCode = "an unexpected error code";
+
+  fetchMock.postOnce(urls.NA + paths.EVENTS, {
+    status: 500,
+    body: {
+      header: {
+        namespace: "System",
+        name: "Exception",
+        messageId: "a message id"
+      },
+
+      payload: {
+        code: errorCode,
+        description: "description of the error"
+      }
+    }
+  });
+
+  await unitUnderTest
+    .sendTextMessageEvent(userRequestToAlexa, access_token)
+    .then(alexaResponse => {
+      expect(alexaResponse).toEqual(
+        cannedErrorResponses.get(customErrorCodes.UNKNOWN_ERROR)
+      );
+    });
+});
+
+it("handles gracefully when the http response is 'ok' and then we fail to parse the text body out of it.", async () => {
+  const userRequestToAlexa = "user request to Alexa";
+  const access_token = "a mock access_token";
+
+  fetchMock.postOnce(urls.NA + paths.EVENTS, {
```

Comment:
Use constant EVENTS_URL from AVSGateway in the tests, wherever applicable.

Reply:
Done.